src/expandableCalendar/WeekCalendar/index.tsx

```diff
@@ -1,221 +1,225 @@
-      const d = this.getDate(index);
-      array.push(d);
-    }
-    return array;
-  }
+  const containerWidth = useMemo(() => {
+    return calendarWidth ?? constants.screenWidth;
+  }, [calendarWidth]);
 
-  getDate(weekIndex: number) {
-    const {current, context, firstDay = 0} = this.props;
-    const d = new XDate(current || context.date);
-    // get the first day of the week as date (for the on scroll mark)
-    let dayOfTheWeek = d.getDay();
-    if (dayOfTheWeek < firstDay && firstDay > 0) {
-      dayOfTheWeek = 7 + dayOfTheWeek;
+  const onDayPressCallback = useCallback((value?: DateData) => {
+    if (onDayPress) {
+      onDayPress(value);
+    } else if (value) {
+      setDate?.(value.dateString, UpdateSources.DAY_PRESS);
     }
+  }, [onDayPress]);
 
-    // leave the current date in the visible week as is
-    const dd = weekIndex === 0 ? d : d.addDays(firstDay - dayOfTheWeek);
-    const newDate = dd.addWeeks(weekIndex);
-    return toMarkingFormat(newDate);
-  }
-
-  getWeekStyle = memoize((width, style) => {
-    return [{width}, style];
-  });
+  const getWeekStyle = useMemo(() => {
+    return [{width: containerWidth}, propsStyle];
+  }, [containerWidth, propsStyle]);
 
-  onDayPress = (value: DateData) => {
-    this.presenter.onDayPress(this.props.context, value);
-  };
-
-  onScroll = (event: NativeSyntheticEvent<NativeScrollEvent>) => {
-    const x = event.nativeEvent.contentOffset.x;
-    const {onScroll} = this.presenter;
-    const {context} = this.props;
-    const {items} = this.state;
-    const {containerWidth, page} = this;
-
-    const updateState = (newData: string[], newPage: number) => {
-      this.page = newPage;
-      this.setState({items: [...newData]});
-    };
-
-    onScroll({context, updateState, x, page, items, width: containerWidth});
-  };
-
-  onMomentumScrollEnd = () => {
-    const {items} = this.state;
-    const {onMomentumScrollEnd} = this.presenter;
-
-    const updateItems = (items: string[]) => {
-      setTimeout(() => {
-        this.setState({items: [...items]});
-      }, 100);
-    };
-
-    onMomentumScrollEnd({items, props: this.props, page: this.page, updateItems});
-  };
-
-  renderItem = ({item}: any) => {
-    const {allowShadow, context, ...calendarListProps} = this.props;
-    const {style, onDayPress = this.onDayPress, firstDay = 0, ...others} = extractCalendarProps(calendarListProps);
-
-    const isSameWeek = sameWeek(item, context.date, firstDay);
-    const currentContext = isSameWeek ? context : undefined;
+  const renderItem = useCallback(({item}: {item: string}) => {
+    const currentContext = sameWeek(date, item, firstDay) ? context : undefined;
 
     return (
       <Week
         {...others}
         key={item}
         current={item}
         firstDay={firstDay}
-        style={this.getWeekStyle(this.containerWidth, style)}
-        onDayPress={onDayPress}
+        style={getWeekStyle}
```

Comment:
Let's change the name from `getWeekStyle` to `WeekStyle` since it's memoized const and not a method.